Trip.py

```diff
@@ -30,15 +32,39 @@ def _build_legs(self):
             else:
                 self._legs[i] = Leg(qp, [segment['travel_mode']])
 
+    def _find_best_mode(self):
+        for mode in self._acceptable_modes:
+            qp = QueryParser(self._query_parser.get_start(),
```

Comment:
Sort of. We initialize a QueryParser for the first mode in acceptable_modes. Then if there are more acceptable, we create a new QueryParser, qp,  for each mode and update self._query_parser if and only if qt's duration is less than self._query_parsers. 

After iterating through each acceptable mode, we will have the fastest overall trip that Google can provide. But we still have to dig through each segment of the trip and build the legs. This is sort-of implemented but still buggy.

## Side issue:

I noticed an issue in the constructor - we should set the legs regardless of how many acceptable modes there are.